samples/dnn/edge_detection.cpp

```diff
@@ -0,0 +1,216 @@
+        "For switching between deep learning based model(dexined) and canny edge detector, press 'd' (for dexined) or 'c' (for canny) respectively.\n\n"
+        "Script is based on https://github.com/axinc-ai/ailia-models/blob/master/line_segment_detection/dexined/dexined.py\n"
+        "To download the onnx model, see: https://storage.googleapis.com/ailia-models/dexined/model.onnx"
+        "\n\nOpenCV onnx importer does not process dynamic shape. These need to be substituted with values using:\n\n"
+        "python3 -m onnxruntime.tools.make_dynamic_shape_fixed --dim_param w --dim_value 640 model.onnx model.sim1.onnx\n"
+        "python3 -m onnxruntime.tools.make_dynamic_shape_fixed --dim_param h --dim_value 480 model.sim1.onnx model.sim.onnx\n";
+
+    const string param_keys =
+        "{ help h          |            | Print help message. }"
+        "{ input i         | baboon.jpg | Path to the input image }"
+        "{ model           |            | Path to the ONNX model. Required. }"
+        "{ imageSize       |   512      | Image Size}";
+
+    const string backend_keys = format(
+        "{ backend         | 0 | Choose one of computation backends: "
+        "%d: automatically (by default), "
+        "%d: Intel's Deep Learning Inference Engine (https://software.intel.com/openvino-toolkit), "
+        "%d: OpenCV implementation, "
+        "%d: VKCOM, "
+        "%d: CUDA, "
+        "%d: WebNN }",
+        cv::dnn::DNN_BACKEND_DEFAULT, cv::dnn::DNN_BACKEND_INFERENCE_ENGINE, cv::dnn::DNN_BACKEND_OPENCV,
+        cv::dnn::DNN_BACKEND_VKCOM, cv::dnn::DNN_BACKEND_CUDA, cv::dnn::DNN_BACKEND_WEBNN);
+
+    const string target_keys = format(
+        "{ target          | 0 | Choose one of target computation devices: "
+        "%d: CPU target (by default), "
+        "%d: OpenCL, "
+        "%d: OpenCL fp16 (half-float precision), "
+        "%d: VPU, "
+        "%d: Vulkan, "
+        "%d: CUDA, "
+        "%d: CUDA fp16 (half-float preprocess) }",
+        cv::dnn::DNN_TARGET_CPU, cv::dnn::DNN_TARGET_OPENCL, cv::dnn::DNN_TARGET_OPENCL_FP16,
+        cv::dnn::DNN_TARGET_MYRIAD, cv::dnn::DNN_TARGET_VULKAN, cv::dnn::DNN_TARGET_CUDA,
+        cv::dnn::DNN_TARGET_CUDA_FP16);
+
+    const string keys = param_keys + backend_keys + target_keys;
+
+    CommandLineParser parser(argc, argv, keys);
+    parser.about(about);
+
+    string inputImagePath = parser.get<string>("input");
+    string onnxModel = parser.get<string>("model");
+    int backend = parser.get<int>("backend");
+    int target = parser.get<int>("target");
+    int imageSize = parser.get<int>("imageSize");
+
+    if (parser.has("help")) {
+        parser.printMessage();
+        return 0;
+    }
+
+    Mat image = imread(samples::findFile(inputImagePath));
+    if (image.empty()) {
+        cout << "Could not read the image: " << inputImagePath << endl;
+        return 1;
+    }
+
+    string method = "dexined";
```

Comment:
Let's use Canny by default, if model is not provided. Also I propose to handle the case in run-time, e.g. show warning, but not fail.